src/main/java/net/earthcomputer/musicalcode/MemberPattern.java

```diff
@@ -52,20 +52,24 @@ public static MemberPattern parse(File file, Remapper yarn2intermediary) {
     }
 
     public static MemberPattern parse(String pattern, Remapper yarn2Intermediary) {
-        int dotIndex = pattern.indexOf('.');
-        if (dotIndex == -1) {
+        int semicolonIndex = pattern.indexOf(';');
+        if (semicolonIndex == -1) {
             if (!INTERNAL_NAME_PATTERN.matcher(pattern).matches()) {
                 throw new IllegalArgumentException(pattern + " does not match the pattern for class names");
             }
             return new ClassPattern(yarn2Intermediary.map(pattern));
         }
 
-        String className = pattern.substring(0, dotIndex);
-        String member = pattern.substring(dotIndex + 1);
+        String className = pattern.substring(0, semicolonIndex);
+        // remove L from Lnet to allow remapping. Allow \ as escape character
+        if (className.startsWith("Lnet") || className.startsWith("\\")) {
```

Comment:
I wouldn't hardcode matching for `Lnet`. If there is a semicolon in the pattern, it makes sense to also require an `L` at the start, otherwise do no special handling for the `L`. If we want to be backwards compatible with the old format, we can check for the presence of `.` in the string.